apps/sms/js/activity_handler.js

```diff
@@ -294,70 +293,56 @@ var ActivityHandler = {
       };
 
     if (Compose.isEmpty()) {
-      this.launchComposer(activity);
-    } else {
-      // ask user how should we do
-      ActivityHandler.displayUnsentConfirmation(activity);
+      return this.launchComposer(activity);
     }
+
+    // ask user how should we do
+    ActivityHandler.displayUnsentConfirmation(activity);
   },
-  // Deliver the user to the correct view
-  // based on the params provided in the
-  // "message" object.
-  //
-  toView: function ah_toView(message) {
-    /**
-     *  "message" is either a message object that belongs
-     *  to a thread, or a message object from the system.
-     *
-     *
-     *  message {
-     *    number: A string phone number to pre-populate
-     *            the recipients list with.
-     *
-     *    body: An optional body to preset the compose
-     *           input with.
-     *
-     *    contact: An optional "contact" object
-     *
-     *    threadId: An option threadId corresponding
-     *              to a new or existing thread.
-     *
-     *  }
-     */
 
+  /**
+   * Delivers the user to the correct view based on the params provided in the
+   * "message" parameter.
+   * @param {{number: string, body: string, contact: MozContact,
+   * threadId: number}} message It's either a message object that belongs to a
+   * thread, or a message object from the system. "number" is a string phone
+   * number to pre-populate the recipients list with, "body" is an optional body
+   * to preset the compose input with, "contact" is an optional MozContact
+   * instance, "threadId" is an optional threadId corresponding to a new or
+   * existing thread.
+   * @returns {Promise}
+   */
+  toView: function ah_toView(message) {
     if (!message) {
-      return;
+      return Promise.reject(new Error('Message is not defined!'));
```

Comment:
yeah, you're right

I also tried to check the case of clicking on the notification for a message that has been deleted -- but this should not happen now because we remove the notification when we enter or delete a thread :) (unless we have another bug somewhere, so we can keep the check -- I mean the getMessage call in handleMessageNotification)